src/palace/manager/api/overdrive/api.py

```diff
@@ -739,25 +753,34 @@ def patron_request(
             # self.log.debug("%s: %s", url, response.status_code)
             return response
 
-    def get_patron_credential(
-        self, patron: Patron, pin: str | None, is_fulfillment: bool = False
+    def _get_patron_oauth_credential(
+        self, patron: Patron, pin: str | None, palace_context: bool = False
     ) -> Credential:
-        """Create an OAuth token for the given patron.
+        """Get an Overdrive OAuth token for the given patron.
+
+        See: https://developer.overdrive.com/apis/patron-auth
 
         :param patron: The patron for whom to fetch the credential.
         :param pin: The patron's PIN or password.
-        :param is_fulfillment: Boolean indicating whether we need a fulfillment credential.
+        :param palace_context: Determines if the oauth token is fetched
+           using the palace credentials or the collections credentials.
         """
 
-        def refresh(credential: Credential) -> Credential:
-            return self.refresh_patron_access_token(
-                credential, patron, pin, is_fulfillment=is_fulfillment
-            )
+        refresh = partial(
+            self._refresh_patron_oauth_token,
+            patron=patron,
+            pin=pin,
+            palace_context=palace_context,
+        )
 
         return Credential.lookup(
             self._db,
             DataSource.OVERDRIVE,
-            "Fulfillment OAuth Token" if is_fulfillment else "OAuth Token",
+            (
+                "Palace Scope OAuth Token"
+                if palace_context
+                else "Collection Scope OAuth Token"
+            ),
```

Comment:
This will force new tokens everywhere when this deployed, but tokens only last an hour, so its not a huge deal. Since all these tokens have expiry, they will all eventually expire and be reaped.